app/models/locations/map.rb

```diff
@@ -0,0 +1,59 @@
+module Locations
+  class Map
+    include ActiveModel::Model
+    validates :loc, presence: true
+
+    def initialize(params = {})
+      @@locator_url = "http://library.princeton.edu/locator/index.php?"
+      @@stackmap_url = "http://princeton.stackmap.com/view/?"
+      @params = params
+    end
+
+    def url
+      if @params[:loc] && @params[:id]
+        holding_location = get_holding_location(@params[:loc])
+        bibrec = get_bibrec(@params[:id])
+        if !holding_location.nil? && !bibrec.nil?
```

Comment:
Again, adding these to a guard statement like above would break this branch.

https://www.youtube.com/watch?v=T8J0j2xJFgQ and the associated book (which I haven't read) has some good resources about these patterns.

It's in the Refactoring book by Fowler too (http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)